### ISSUE TITLE
csg-tutorial/propane/imc does not run through

### ISSUE THREAD

**Issue by marvinbernhardt** [votca/csg#701]
I am not sure what is happening here. The setup is to update alternatively A-A, B-B, A-B.
As some point the potentials become non-physical and consequently the simulation in step_008 crashes.

I'm using current master branch with gromacs 2019.6. I guess this worked at some point. Which version of votca should I try?

A-A after step 0 (k), 1 (r), 4 (g), 7 (b)
![image](https://user-images.githubusercontent.com/25516935/126274790-4119f960-e804-42ab-b824-1f4d089b9398.png)

B-B after step 0 (k), 2 (r), 5 (g)
![image](https://user-images.githubusercontent.com/25516935/126275022-b00374be-7e50-4e18-b789-f039096cd73f.png)

A-B after step 0 (k), 3 (r), 6 (g)
![image](https://user-images.githubusercontent.com/25516935/126275217-f3f5352c-c9b5-4b9e-9eb4-7e3dcc7bb00e.png)

**Comment by marvinbernhardt** [comment]
@DRosen285 any idea?

**Comment by DRosen285** [comment]
It seems that sth. went wrong already after the 1st update of B-B. The potential well is about -25 (if I interpret the scales correctly).  That is too attractive. This might indicate that the initial guess is too bad. What you could try is to run the IBI example first for a couple of iterations and start from there. Or you could add some regularization. I'm trying to reproduce, but I can only run simulations on my laptop at the moment, which is super slow.

**Comment by marvinbernhardt** [comment]
It is also super slow on my computer^^. I agree the B-B update looks bad. I also tried updating all potentials at the same step and then it is also B-B which becomes much too attractive.

**Comment by junghans** [comment]
@JensWehner did a big refactor of IMC in v1.6, so it might be worth trying IMC in the 1.5.1 docker container.

**Comment by DRosen285** [comment]
Yeah , the 1.6 container still has the following bug: https://github.com/votca/csg/issues/577.
So 1.5.1 is a better choice.

**Comment by DRosen285** [comment]
@marvinbernhardt you could also try to increase  r_0 from 0.31 to 0.32 or 0.33  for the B-B interaction.
I did some tests with shorter trajectories and that has improved the B-B potential, but it is still pretty negative, but there is now a larger repulsive core.

**Comment by marvinbernhardt** [comment]
I'm running it now with 1.5.1, let's see. Interestingly votca uses gmx_d.

**Comment by marvinbernhardt** [comment]
Ok new evidence: The docker container with votca v1.5.1 yields the pretty similar RDFs

<details>
  <summary>All three RDFs</summary>

![image](https://user-images.githubusercontent.com/25516935/126497451-8018c03c-11ba-4286-8f23-93f3c6fbe757.png)
</details>

but gives a different A-A potential update at the first step

<details>
  <summary>A-A potentials after first step</summary>

black -> master, red -> v1.5.1
![image](https://user-images.githubusercontent.com/25516935/126497152-f4959c0f-0cbd-49cb-9b7c-6e1e663a3cca.png)
</details>

**Comment by marvinbernhardt** [comment]
It seems like the correlation matrix is “scaled down" for master compared to v1.5.1.
<details>
  <summary>v1.5.1</summary>

![image](https://user-images.githubusercontent.com/25516935/126500192-a22094da-8423-46cc-ac96-e0cd0c307f64.png)
</details>

<details>
  <summary>master</summary>

![image](https://user-images.githubusercontent.com/25516935/126500279-ff70c428-82ef-4c7f-b52c-68eb5b9a4fcc.png)
</details>

Not sure if v1.5.1 will converge, it is very slow.

**Comment by marvinbernhardt** [comment]
hmmm with v1.5.1 it also crashes after B-B becomes to attractive in step 2.

**Comment by DRosen285** [comment]
In the original VOTCA paper there is no data for IMC and propane. So not sure, how well that example actually worked. 
Maybe @junghans knows more.
So I'd try to play with the cut-off, or add regularization, or perform some steps of IBI first or do a combination of all the things. Increasing the sampling could also help, but given that it is already slow, that's maybe the least feasible option.

**Comment by junghans** [comment]
Yeah, I looked in a couple of place, but I don't see any IMC results anywhere, so this tutorial might have never worked...

**Comment by marvinbernhardt** [comment]
> It seems like the correlation matrix is “scaled down" for master compared to v1.5.1.

I did some tests with v1.5.1 but using the same trajectory to see where the scaling arises. There seems to be no bug in votca, rather just different correlations in the trajectories generated. I'm still a bit puzzled to be honest.

**Comment by junghans** [comment]
Looking at https://github.com/votca/csg-tutorials/pull/140, did you find it?

**Comment by marvinbernhardt** [comment]
Not really, no. The current master code seems to converge the propane/imc case after some first steps using IBI.

**Comment by junghans** [comment]
> Not really, no. The current master code seems to converge the propane/imc case after some first steps using IBI.

Ah ok, you could provide `<name>.pot.in` in the tutorials that comes from a `ibi` `pre/` step.
